x-pack/legacy/plugins/siem/public/components/timeline/search_super_select/index.tsx

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiHighlight,
+  EuiInputPopover,
+  EuiSuperSelect,
+  EuiSelectable,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTextColor,
+  EuiFilterButton,
+  EuiFilterGroup,
+  EuiSpacer,
+} from '@elastic/eui';
+import { Option } from '@elastic/eui/src/components/selectable/types';
+import React, { memo, useCallback, useMemo, useState } from 'react';
+import { ListProps } from 'react-virtualized';
+import { createGlobalStyle } from 'styled-components';
+
+import { AllTimelinesQuery } from '../../../containers/timeline/all';
+import { getEmptyTagValue } from '../../empty_value';
+import { isUntitled } from '../../../components/open_timeline/helpers';
+import * as i18nTimeline from '../../../components/open_timeline/translations';
+import { SortFieldTimeline, Direction } from '../../../graphql/types';
+
+import * as i18n from './translations';
+
+const SearchTimelineSuperSelectGlobalStyle = createGlobalStyle`
+  .euiPopover__panel.euiPopover__panel-isOpen.timeline-search-super-select-popover__popoverPanel {
+    visibility: hidden;
+    z-index: 0;
+  }
+`;
+
+interface SearchTimelineSuperSelectProps {
+  isDisabled: boolean;
+  timelineId: string | null;
+  timelineTitle: string | null;
+  onTimelineChange: (timelineTitle: string, timelineId: string) => void;
+}
+
+const basicSuperSelectOptions = [
+  {
+    value: '-1',
+    inputDisplay: i18n.DEFAULT_TIMELINE_TITLE,
+  },
+];
+
+const basicSelectableOptions = [
+  {
+    description: i18n.DEFAULT_TIMELINE_DESCRIPTION,
+    label: i18n.DEFAULT_TIMELINE_TITLE,
+    id: null,
+    title: i18n.DEFAULT_TIMELINE_TITLE,
+  } as Option,
+];
```

Comment:
nit: Consider extending the EUI Option instead of `as` casting:

e.g.
``` ts
interface TimelineSuperSelectableOption extends Option {
  id: string | null;
  title: string;
  description: string;
}

const basicSelectableOptions: TimelineSuperSelectableOption[] = [
  {
    description: i18n.DEFAULT_TIMELINE_DESCRIPTION,
    label: i18n.DEFAULT_TIMELINE_TITLE,
    id: null,
    title: i18n.DEFAULT_TIMELINE_TITLE,
  }
```

Then you can use this type when declaring your callbacks as well which stomps out some `any`'s and provides better type checking. E.g.

``` ts
  const handleTimelineChange = useCallback((options: TimelineSuperSelectableOption[]) => {
    const selectedTimeline = options.filter(option => option.checked === 'on');
    if (selectedTimeline != null && selectedTimeline.length > 0 && onTimelineChange != null) {
      onTimelineChange(selectedTimeline[0].title, selectedTimeline[0].id ?? '');
    }
    setIsPopoverOpen(false);
  }, []);
```